src/components/SingleColor/index.jsx

```diff
@@ -1,77 +1,100 @@
+import React from "react";
+import "./SingleColor.css";
+
+class SingleColor extends React.Component {
+  state = { alert: false };
+
+  // ?????
+  //   useEffect(() => {
+  //     const timeout = setTimeout(() => {
+  //       setAlert(false);
+  //     }, 300000);
+  //     return () => clearTimeout(timeout);
+  //   }, [alert]);
+
+  // USAGE
+  // componentDidUpdate(prevProps) {
+  //   // Typical usage (don't forget to compare props):
+  //   if (this.props.userID !== prevProps.userID) {
+  //     this.fetchData(this.props.userID);
+  //   }
+  // }
+
+  // componentDidUpdate(prevProps, prevState) {
+  //   if (prevState.pokemons !== this.state.pokemons) {
+  //     console.log('pokemons state has changed.')
+  //   }
+  // }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { alert } = this.state;
+
+    if (prevState.alert !== alert) {
+      this.timeout = setTimeout(() => {
+        this.setState({ alert: false });
+      }, 3000);
+    }
```

Comment:
this looks right. apres!

Reply:
I couldn't access it, when it was assigned to a variable, how can I do it?

Reply:
also my alert state isn't updating when I want it to update, what am I doing wrong? added comment to the file